mkosi/distributions/gentoo.py

```diff
@@ -16,56 +17,91 @@
 from mkosi.remove import unlink_try_hard
 from mkosi.run import run, run_workspace_command
 from mkosi.state import MkosiState
+from mkosi.types import PathString
+
+thread_counts = (os.cpu_count() or 1) * 2
 
 
 def invoke_emerge(
     state: MkosiState,
+    sysroot: Optional[Path] = None,
+    root: Optional[Path] = None,
+    bwrap_params: list[PathString] = [],
     pkgs: Sequence[str] = (),
     actions: Sequence[str] = (),
     opts: Sequence[str] = (),
+    env: dict[str, str] = {},
 ) -> None:
-    jobs = os.cpu_count() or 1
+    bwrap: list[PathString] = []
+    if sysroot is not None:
+        # this is inside our ns for stage, i.e. this is NOT host's /tmp
+        target_root_mntp = "/tmp/mkosi-root"
```

Comment:
This could use a lot more information on why this is necessary. I'm confused why we need both `root` and `sysroot`?

Reply:
`root` is where we're installing, `sysroot` is our toolchain (portage, compiler, linker etc).
See here for more: [emerge(1)](https://dev.gentoo.org/~zmedico/portage/doc/man/emerge.1.html).